lib/sla.js

```diff
@@ -128,13 +186,58 @@ function updateState(client, name, state) {
  * @param {Object} inbound- the inbound channel
  * @param {Object} bridge - Object that defines the bridge to be passed to
  *   addChannelsToBridge().
- * @ param {Object} trunkData - the data of the trunk to be used for station
- *   access.
+ * @param {Object} extension - the data structure that represents the extension,
+ *   mainly used for telling whether or not an extension is busy or not
  * @return {Q} Q promise object.
  */
-function originateChannel(client, inbound, bridge, trunkData) {
-  var defer = Q.defer();
+function originateInbound(client, inbound, bridge, extension) {
+
   var dialed = [];
+  var defer = Q.defer();
+
+  getState(client, bridge.name).then(function(currentState) {
+
+    if (currentState !== inUse && currentState !== busy &&
+        currentState !== ringing) {
+      extension.getAllStations()
+        .then(function (stations) {
+
+          if (stations.length) {
+            stations.forEach(function(station) {
```

Comment:
I believe dialed is used to keep track of stations/channels throughout the lifetime of the application for a given extension.

Comments would definitely help here.